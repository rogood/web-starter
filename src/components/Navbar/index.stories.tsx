import React from 'react'
import Navbar from './'

export default {
    title: 'Navbar',
}

export const navbar = (): JSX.Element => (
    <Navbar rightContent={<a href="/">Log In</a>} />
)
