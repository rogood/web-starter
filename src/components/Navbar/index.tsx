import React from 'react'
import { Container, Dropdown, Image, Menu } from 'semantic-ui-react'
import useAuth from '../../hooks/useAuth'
import useErrorToast from '../../hooks/useErrorToast'
import Router from 'next/router'
import { signOut } from '../../api/auth'

export default function Navbar(): JSX.Element {
    const user = useAuth(null, { redirectIfNoUser: false })
    const { showErrorToast } = useErrorToast()

    async function onSignOutClick(): Promise<void> {
        try {
            await signOut()
            Router.push('/')
        } catch (error) {
            showErrorToast(error)
        }
    }

    return (
        <div>
            <Menu fixed="top" inverted>
                <Container>
                    <Menu.Item as="a" header href="/">
                        <Image
                            size="tiny"
                            alt="List It Logo"
                            src="/images/logo-inverse.png"
                            style={{ marginRight: '1.5em' }}
                        />
                    </Menu.Item>
                    <Menu.Menu position="right">
                        <Dropdown
                            item
                            simple
                            text={user ? user.displayName : 'Log In'}
                        >
                            <Dropdown.Menu>
                                <Dropdown.Item onClick={onSignOutClick}>
                                    Sign out
                                </Dropdown.Item>
                            </Dropdown.Menu>
                        </Dropdown>
                    </Menu.Menu>
                </Container>
            </Menu>
        </div>
    )
}
