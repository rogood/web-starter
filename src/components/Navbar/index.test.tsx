import React from 'react'
import Navbar from './index'
import { render } from '@testing-library/react'

describe('Navbar', () => {
    test('matches screenshot', () => {
        const { container } = render(<Navbar />)
        expect(container).toMatchSnapshot()
    })
})
