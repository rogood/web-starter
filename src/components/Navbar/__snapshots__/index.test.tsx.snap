// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches screenshot 1`] = `
<div>
  <div>
    <nav>
      <div
        class="logo-content"
      >
        <a
          href="/"
        >
          <img
            alt="List It Logo"
            src="images/logo-inverse.png"
          />
        </a>
      </div>
      <div
        class="right-content"
      />
    </nav>
    <nav
      class="ui inverted top fixed menu"
    >
      <div
        class="ui container"
      >
        <a
          class="header item"
          href="/"
        >
          <img
            class="ui tiny image"
            src="images/logo-inverse.png"
            style="margin-right: 1.5em;"
          />
        </a>
        <div
          class="right menu"
        >
          <div
            aria-expanded="false"
            class="ui item simple dropdown"
            role="listbox"
            tabindex="0"
          >
            <div
              aria-atomic="true"
              aria-live="polite"
              class="text"
              role="alert"
            >
              Log In
            </div>
            <i
              aria-hidden="true"
              class="dropdown icon"
            />
            <div
              class="menu transition"
            >
              <div
                class="item"
                role="option"
              >
                List Item
              </div>
              <div
                class="item"
                role="option"
              >
                List Item
              </div>
              <div
                class="divider"
              />
              <div
                class="header"
              >
                Header Item
              </div>
              <div
                class="item"
                role="option"
              >
                <i
                  class="dropdown icon"
                />
                <span
                  class="text"
                >
                  Submenu
                </span>
                <div
                  class="menu transition"
                >
                  <div
                    class="item"
                    role="option"
                  >
                    List Item
                  </div>
                  <div
                    class="item"
                    role="option"
                  >
                    List Item
                  </div>
                </div>
              </div>
              <div
                class="item"
                role="option"
              >
                List Item
              </div>
            </div>
          </div>
        </div>
      </div>
    </nav>
    <style>
      
                .nav {
                    display: flex;
                    background-color: #37A6CD;
                    height: 32px;
                    width: calc(100vw - 32px);
                    padding: 16px;
                    align-items: center;
                    justify-content: space-between;
                    color: #FFF;
                }
                .logo-content {
                }
                .logo-content img {
                    height: 32px;
                }
                .right-content a {
                    color: #FFF;
                    text-decoration: none;
                }
            
    </style>
  </div>
</div>
`;
